int32 leftEncoder
int32 rightEncoder